BDArmory/Ammo/PooledBullet.cs

```diff
@@ -506,9 +509,21 @@ void FixedUpdate()
         /// <param name="period">Period to consider, typically Time.fixedDeltaTime</param>
         public void MoveBullet(float period)
         {
+            atmosphereDensity = (float)FlightGlobals.getAtmDensity(FlightGlobals.getStaticPressure(currentPosition), FlightGlobals.getExternalTemperature(currentPosition)); 
             // Initial half-timestep velocity change (leapfrog integrator)
             LeapfrogVelocityHalfStep(0.5f * period);
 
+            if (targetVessel != null && atmosphereDensity > 0.05f)
+            {
+                if (penTicker == 0 && Vector3.Dot((targetVessel.CoM - currentPosition).normalized, currentVelocity.normalized) > 0) //don't circle around if it misses, or after it hits something
```

Comment:
None of the vectors need normalising here as you're only checking the sign of the dot product and `maxMagnitudeDelta` is 0 in the `RotateTowards` line.
There should also be a factor of `period` in the `maxRadiansDelta` argument of `RotateTowards`.